### DISCUSSION TITLE
Docker Private Registry on separate host

### DISCUSSION THREAD

**Discussion by collegeimprovements** [BretFisher/ama#208] [Q&A]
Hello Bret, 

Your Udemy course is awesome.

Would it be possible for you to create a demo/video/tutorial for `docker private registry` on a different host. All tutorials explain how to setup registry and then tag images and then push/pull those images from same host.
But setting up `private docker registry` on different host requires certificates which seems bit complex.

If that's not possible, any pointer or guideline would be great. 

Thanks a lot :)

**Comment by BretFisher** [comment]
Hi there, good question.

The certificates for registry are no different than using TLS/SSL on other websites. You can buy a web cert and add to your registry, and then you won't need to add it to your docker hosts.

Manual "self-signed" certificates are tedious, if not complex, which is why most people don't use the open source registry by default, and go with a [hosted solution](https://veggiemonk.github.io/awesome-docker/#registry), [Docker EE](https://www.docker.com/enterprise-edition), [GitLab's registry](https://docs.gitlab.com/ce/user/project/container_registry.html), [VMWare's Harbor](http://vmware.github.io/harbor/), [Quay Enterprise](https://quay.io/tour/enterprise), etc.

One option is using the [Portus web GUI](http://port.us.org/) over top of Docker registry, which might make TLS easier.

Did you run through the [PWD process for auth and creating a server cert](https://training.play-with-docker.com/linux-registry-part2/) for the registry? That process is just for those running a small environment or testing that don't need to deal with lots of machines. In the real world, you'd want to go buy a web host cert or use a 3rd party registry.

**Comment by BretFisher** [comment]
keeping this open for others to learn

**Comment by manvindar** [comment]
https://goharbor.io/ is another good option we use, you can scan images and host helm charts also
